
 try{

 const kajal=(10,20)=>{
 var x=10+20;
 if(x==40){

 }
    
 }
    
        
    }




catch(err){


    console.log({name:err.name,message:err.message})
}
finally{
    console.log("its final");
}
// // k();ajal
// kajal()

// let kaju=(x,y)=>x+y;
// try{
// let result=kaju(10,30)
// console.log(result);
// }
// catch(err){
//     console.log({name:err.name , message:err.message});
// }
// catch try...final
// let result=0;
// try{
//     let result=add(10,20)
//     console.log(result);
// }
// catch(err){
//     console.log({name:err.name,message:err.message});
// }
// finally{
//     console.log(result)u
// }

// throw exception in js
// const add=(x,y)=>{
//     if (typeof x!=='number'){
//         throw 'the first argument must be a number'
//     }
//     if (typeof y!=='String'){
//         throw 'the second argument must be a string'
//     }
//     return x+y;
// }
// const result=add("kajal" , 60)
// console.log(result);